### DISCUSSION TITLE
RiverClient Init. Syntax Err

### DISCUSSION THREAD

**Discussion by kashyap2108** [riverqueue/river#509] [Q&A] [answered]
I followd the documentation and wrote a sample code.
```
riverClient, err := river.NewClient(riverpgxv5.New(dbPool), &river.Config{
    Queues: map[string]river.QueueConfig{
        river.QueueDefault: {MaxWorkers: 100},
    },
    Workers: workers,
})
```

but this is showing syntax err.
and i had to change to below with passing a type.
```
riverClient, err := river.NewClient[pgx.Tx](riverpgxv5.New(dbPool), &river.Config{
		Queues: map[string]river.QueueConfig{
			river.QueueDefault: {MaxWorkers: 100},
		},
	})
```

Is there any version issue here ?

**Comment by ciokan** [comment]
Same with the `AddWorker` code. I had to change the official example from `river.AddWorker(workers, &SortWorker{})` to `river.AddWorker[SortArgs](workers, &SortWorker{})`

**Comment by bgentry** [comment]
Since you didn’t include an actual error message, it’s hard to diagnose the issue. But you can see in the executable examples that the documented forms work just fine on 1.21 and 1.22 (our supported versions): https://github.com/riverqueue/river/blob/ab7f61624fd342d2dae98427ffd5ed1c778bc0e7/example_complete_job_within_tx_test.go#L73

What Go version are you on? Or perhaps you have an editor or linter setup that can’t infer types properly?

**Comment by brandur** [comment, marked as answer]
This is quite similar to https://github.com/riverqueue/river/pull/460, in which it turned out to be a misbehaving GoLand was requiring an explicit type parameter.

**Reply by ciokan** [reply]
indeed it is a Goland thing. What made me have doubts was the fact that the terminal was complaining also, the same IDE terminal.
